### DISCUSSION TITLE
filter without filterableAttributes

### DISCUSSION THREAD

**Discussion by shimsag** [meilisearch/product#263] [Feedback & Feature Proposal]
Hi

It would be good if we can filter dynamically on fields not set in filterableAttributes ... As it requires to reload the whole index if we want to add filterableAttributes

**Comment by gmourier** [comment]
Hello @shimsag, thanks for your feedback 👋

Being able to have filtering on fields without determining them first implies computing all data structures for filtered search so that they can be used as-is. 

We have not made this choice for several reasons.

First, we feel that it is necessary to think about the fields that will be filterable by the end-users to establish a good search experience for them.

Limiting the data structure calculations to a specific set of fields also reduces the indexing time and disk size to what is necessary.

If you want to make every field filterable, you can actually list all fields into the filterable attributes when a new field appears.

Finally, this could also be a security issue. By limiting the use of filters to only certain fields chosen by the administrator, you can be sure that it is not possible for the user to filter on fields that are not intended.

Do you have a way to determine at some point which fields will be filterable before going into production? Do you regularly redefine the filterable attributes and if so, why?

Thanks for your answer!
